packages/web3-providers/src/Connection/EVM/apis/Web3API.ts

```diff
@@ -288,9 +293,14 @@ export class Web3API
         // ERC1155
         if (actualSchema === SchemaType.ERC1155) return ''
 
-        // ERC721
-        const contract = this.getERC721Contract(chainId, address)
-        return (await contract?.methods.ownerOf(tokenId).call()) ?? ''
+        try {
+            // ERC721
+            const contract = this.getERC721Contract(chainId, address)
+            return (await contract?.methods.ownerOf(tokenId).call()) ?? ''
+        } catch {
+            const cryptoPunksContract = this.getCryptoPunksContract(chainId, address)
```

Comment:
1. compare the addr if it's punk then use the punk way.
2. user the default erc721 way.